angular2/src/app/app.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, OnInit } from '@angular/core';
+import { ChatEngineCore } from 'chat-engine';
+import { ChatEngine } from './chatEngine';
+declare var require: any;
+const random = require('chat-engine-random-username');
+const search = require('chat-engine-online-user-search');
+
+// console.log(search);
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html'
+})
+export class AppComponent implements OnInit {
+  private ce: any;
+
+  constructor(private chatEngine: ChatEngine) {
+    this.ce = chatEngine;
+  }
+
+  ngOnInit() {
+    this.ce.instance.connect(new Date().getTime(), {}, 'auth-key');
+    this.ce.instance.on('$.ready', (data) => {
+      this.ce.me = data.me;
+      this.ce.me.plugin(random());
+
+      // when I get a private invit
+      this.ce.me.direct.on('$.invite', (payload) => {
+        const chat = new this.ce.instance.Chat(payload.data.channel);
+        chat.onAny((a) => {
+          console.log(a);
```

Comment:
this logs everything, probably not good for production

Reply:
yes, you’re right but as this is a demo then I don’t know, it might be ok to the user see what data flow thought of the sdk